### ISSUE TITLE
.tmp folder sometimes not being deleted after run

### ISSUE THREAD

**Issue by cvoegele** [i4Ds/Karabo-Pipeline#121]

**Comment by deiruch** [comment]
@Lukas113 Is this still a problem?

**Comment by Lukas113** [comment]
@kenfus how is the behavior of this since the PR https://github.com/i4Ds/Karabo-Pipeline/pull/305 ?

**Comment by kenfus** [comment]
@Lukas113 There is no more auto-deleting of data. It creates a folder where you run your data and you have to delete it manually. However, we offer a support-function to delete this folder and can be called by the user (it's also in the example scripts).

**Comment by deiruch** [comment]
What do we store in the `.tmp` folder anyway? What are the downsides of deleting this folder regularly? How much overhead would that result in?

If possible, I'd prefer a solution that just works, without user involvement.

**Comment by Lukas113** [comment]
Thanks for the answer @kenfus.

This was used to store temporary files created by `Karabo`, which was used for subsequent steps such as imaging from Visibilities. Previously the behavior was that the `.tmp` folder was bound to `FileHandle`. That means, if no reference existed on the `FileHandle` anymore, the `__del__` method was called usually, which led to the fact that temporarily created files were deleted automatically. This had the consequence that the `FileHandle` object unnecessarily must already exist and by different steps in the Pipeleine to be passed at (otherwise the folder would have deleted itself). This also meant that you could not add your own already created files (without the risk of deleting the folder, because the `FileHandle` object was mandatory).

The `FileHandle` object created a non-intuitive way of programming in my opinion, where this object had to be passed to the `Image` constructor in `https://github.com/i4Ds/Karabo-Pipeline/blob/main/karabo/imaging/imager.py 150` for example, instead of just the path.

In conclusion, I find @kenfus's solution good. In general we should avoid automatic deletion of files in my opinion, because they could be important for later process steps or the user. Therefore I think we can close this issue.

**Comment by kenfus** [comment]
As discussed with @deiruch , we can close this.
